### PR TITLE
Bump Spack and DLA-Future in CI

### PR DESCRIPTION
**Pull request by RMeli**
Bump Spack to more recent commit and DLA-Future to `dla-future@0.4.0`.

### CODE REVIEW

**Comment by oschuett** [comment]
What's the state of this PR? Anything I could do to help?

**Comment by RMeli** [comment]
Ah. I did not realize this was failing. I'll look into it.

**Comment by oschuett** [comment]
How should we proceed with this PR?

**Comment by RMeli** [comment]
Sorry, I am working on lifting DLA-Future's Fortran interface out of CP2K into a standalone package so that we can have a slimmer integration and also add the standard and generalised eigensolver for complex types.

I was hoping to tackle this PR with the new integration, however it's taking a bit longer than planned. I'll try to bump the Spack version again and then look more closely into it if the issue persists.

File: tools/docker/Dockerfile.test_cmake [hunk 1]
```diff
@@ -44,7 +44,7 @@ RUN apt-get update -qq && apt-get install -qq --no-install-recommends \
 WORKDIR /opt/spack
 RUN git init --quiet && \
     git remote add origin https://github.com/spack/spack.git && \
-    git fetch --quiet --depth 1 origin 8bcb1f8766ffbf097ec685bdafad53bc8b6e9e30 && \
+    git fetch --quiet --depth 1 origin 2721b4c10d27afde281ea0cdb0480a7343ce0863 && \
```
Comment (RMeli) [inline tools/docker/Dockerfile.test_cmake]:
```suggestion
    git fetch --quiet --depth 1 origin 384e1983041094c8487d394f59dfe8a6848d60f2 && \
```

File: tools/docker/generate_dockerfiles.py [hunk 2]
```diff
@@ -779,7 +779,7 @@ def spack_env_toolchain() -> str:
 WORKDIR /opt/spack
 RUN git init --quiet && \
     git remote add origin https://github.com/spack/spack.git && \
-    git fetch --quiet --depth 1 origin 8bcb1f8766ffbf097ec685bdafad53bc8b6e9e30 && \
+    git fetch --quiet --depth 1 origin 2721b4c10d27afde281ea0cdb0480a7343ce0863 && \
```
Comment (RMeli) [inline tools/docker/generate_dockerfiles.py]:
```suggestion
    git fetch --quiet --depth 1 origin 384e1983041094c8487d394f59dfe8a6848d60f2 && \
```

**Comment by RMeli** [comment]
This is more annoying than I thought... I'll work locally with the Docker container and push a solution when it's ready.

**Comment by RMeli** [comment]
I confirmed my suspicion that the problem appears after the introduction of the `gcc-runtime` package. On install, this packages copies the binaries internally, so the warning is spurious.

I think I never saw this locally because I'm using `spack build-env` instead of activating an environment with the default view. It is still not clear to me how to best fix this within the constraints discussed in https://github.com/cp2k/cp2k/pull/3149 and https://github.com/cp2k/cp2k/pull/3154. I'll have another look at this next week.

In the meantime, an ugly workaround is to remove the `libgomp.so` copy made by `gcc-runtime`.

**Comment by RMeli** [comment]
Our local Spack aficionado @albestro helped me to investigate this issue, and he suggested to exclude `gcc-runtime` from the environment view. This prevents `libgomp.so` to be added to the view and therefore CMake only finds the original library in `/usr/lib/gcc/x86_64-linux-gnu/11`.

**Comment by oschuett** [comment]
It seems odd that Spack still builds its own gcc-runtime after it picked up the external gcc. In any case, I'm glad you found a good workaround.

**Comment by albestro** [comment]
> It seems odd that Spack still builds its own gcc-runtime after it picked up the external gcc. In any case, I'm glad you found a good workaround.

Just for clarity, spack is not "building" a `gcc-runtime`, indeed it is copying the compiler libraries needed for runtime from the compiler you are using. This is useful in case the compiler is removed, but whatever has been built will still work correctly.

The warning you get in the CMake is a legit warning, because it is not checking that those libraries are actually the same, but we know they are.

**Pull request merged**